packages/ai/core/embed/embed-many.ts

```diff
@@ -100,7 +100,17 @@ Only applicable for HTTP-based providers.
     }),
     tracer,
     fn: async span => {
-      const maxEmbeddingsPerCall = await model.maxEmbeddingsPerCall;
+      const [maxEmbeddingsPerCall, supportsParallelCalls] = await Promise.all([
+        model.maxEmbeddingsPerCall,
+        model.supportsParallelCalls,
+      ]);
+
+      const maxParallelChunks =
+        typeof supportsParallelCalls === 'number'
+          ? supportsParallelCalls
+          : supportsParallelCalls
+            ? Infinity
+            : 1;
```

Comment:
Why do we need to cap the max tool calls based on the provider? for which provider would this matter vs boolean?